astropy/coordinates/tests/test_intermediate_transformations.py

```diff
@@ -270,6 +295,33 @@ def test_gcrs_altaz():
     assert_allclose(aa1.az, aa3.az)
 
 
+def test_gcrs_hadec():
+    """
+    Check GCRS<->HADec transforms for round-tripping.  Has multiple paths
+    """
+    from astropy.coordinates import EarthLocation
+
+    usph = golden_spiral_grid(128)
+    gcrs = GCRS(usph, obstime='J2000')  # broadcast with times below
+
+    # check array times sure N-d arrays work
+    times = Time(np.linspace(2456293.25, 2456657.25, 51) * u.day,
+                 format='jd')[:, np.newaxis]
+
+    loc = EarthLocation(lon=10 * u.deg, lat=80. * u.deg)
+    hdframe = HADec(obstime=times, location=loc)
+
+    hd1 = gcrs.transform_to(hdframe)
+    hd2 = gcrs.transform_to(ICRS()).transform_to(CIRS()).transform_to(hdframe)
+    hd3 = gcrs.transform_to(ITRS()).transform_to(CIRS()).transform_to(hdframe)
+
+    # make sure they're all consistent
+    assert_allclose(hd1.dec, hd2.dec)
+    assert_allclose(hd1.ha, hd2.ha)
+    assert_allclose(hd1.dec, hd3.dec)
+    assert_allclose(hd1.ha, hd3.ha)
```

Comment:
@mkbrewer - to address @StuartLittlefair's larger comment, I'd suggest to here also check the numerical values of one of these three, just to ensure we do not later do something silly that makes, e.g., `HADec` be just an alias for `AltAz` (as is, I think all the tests would pass if one were to do that!).

Reply:
OK. I'll go ahead and create a precision test.